types/src/dkg.rs

```diff
@@ -22,20 +24,69 @@ pub struct DKGTranscriptMetadata {
 /// Reflection of Move type `0x1::dkg::DKGStartEvent`.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct DKGStartEvent {
-    pub target_epoch: u64,
+    pub session_metadata: DKGSessionMetadata,
     pub start_time_us: u64,
-    pub target_validator_set: ValidatorSet,
 }
 
 impl MoveStructType for DKGStartEvent {
     const MODULE_NAME: &'static IdentStr = ident_str!("dkg");
     const STRUCT_NAME: &'static IdentStr = ident_str!("DKGStartEvent");
 }
 
-/// DKG parameters.
+/// Reflection of Move type `0x1::dkg::DKGConfig`.
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
 pub struct DKGConfig {
-    //TODO
+    variant: MoveAny,
+}
+
+impl Default for DKGConfig {
+    fn default() -> Self {
+        Self {
+            variant: DKGConfigV0::default().as_move_any(),
+        }
+    }
+}
+
+impl OnChainConfig for DKGConfig {
+    const MODULE_IDENTIFIER: &'static str = "dkg";
+    const TYPE_IDENTIFIER: &'static str = "DKGConfig";
+}
+
+/// Reflection of Move type `0x1::dkg::DKGConfigV0`.
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
+pub struct DKGConfigV0 {
+    // When this equals to `x`, `x / u64::MAX` is the DKG transcript aggregation threshold.
+    aggregation_threshold: u64,
```

Comment:
How is this used? I thought we are just using `check_voting_power`.

Reply:
in a real dkg this will be used in transcript verification.

Reply:
I thought we use `check_voting_power` in trx verification, no? Also the `x / u64::MAX` definition is a bit strange to me

Reply:
fair, let's parameterize it later when needed